### PR TITLE
[Feature] Support Uncertain L1 Loss

### PR DESCRIPTION
**Pull request by Tai-Wang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
Support uncertain L1 loss and smooth L1 loss used in the KITTI experiments of [PGD](https://arxiv.org/abs/2107.14160).

## Modification
Please briefly describe what modification is made in this PR.
Add uncertain L1 loss and corresponding unit tests.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
No.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/models/losses/uncertain_smooth_l1_loss.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        alpha (float, optional): The coefficient of log(sigma).
+            Defaults to 1.0.
+        beta (float, optional): The threshold in the piecewise function.
+            Defaults to 1.0.
+
+    Returns:
+        torch.Tensor: Calculated loss
+    """
+    assert beta > 0
+    assert pred.size() == target.size() == sigma.size() and \
```
Comment (ZwwWayne) [inline mmdet3d/models/losses/uncertain_smooth_l1_loss.py:27]: assertion message is necessary

File: mmdet3d/models/losses/uncertain_smooth_l1_loss.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import mmcv
+import torch
+from torch import nn as nn
+
+from mmdet.models.builder import LOSSES
+from mmdet.models.losses.utils import weighted_loss
+
+
+@mmcv.jit(derivate=True, coderize=True)
```
Comment (ZwwWayne) [inline mmdet3d/models/losses/uncertain_smooth_l1_loss.py:9]: rm mmcv.jit

File: mmdet3d/models/losses/uncertain_smooth_l1_loss.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+@LOSSES.register_module()
+class UncertainL1Loss(nn.Module):
+    """L1 loss with uncertainty.
+
+    Args:
+        alpha (float, optional): The coefficient of log(sigma).
+            Defaults to 1.0.
+        reduction (str, optional): The method to reduce the loss.
+            Options are "none", "mean" and "sum".
```
Comment (ZwwWayne) [inline mmdet3d/models/losses/uncertain_smooth_l1_loss.py:131]: Default

**Review by ZwwWayne — APPROVED** [review]
